packages/utils/ast-utils.js

```diff
@@ -429,6 +429,62 @@ function addProperty(j, p, key, value) {
 	}
 	return;
 }
+
+/**
+ *
+ * Recursively updates an object/property to a node
+ * @param {any} j — jscodeshift API
+ * @param {Node} p - AST node
+ * @param {String} key - key of a key/val object
+ * @param {Any} value - Any type of object
+ * @returns {Node} - the updated ast
+ */
+
+function updateProperty(j, p, key, value) {
+	let valForNode;
+	if (!p) {
+		return;
+	}
+	if (Array.isArray(value)) {
+		let arrExp = j.arrayExpression([]);
+		if (safeTraverse(p, [
+			"value",
+			"value",
+		]) && p.value.value.type === "ArrayExpression") {
+			arrExp = p.value.value;
+		}
+		value.forEach(val => {
+			addProperty(j, arrExp, null, val);
+		});
+		valForNode = arrExp;
+	} else if (
+		typeof value === "object" &&
+		!(value.__paths || value instanceof RegExp)
+	) {
+		let objectExp = j.objectExpression([]);
+		if (safeTraverse(p, [
+			"value",
+			"value",
+		]) && p.value.value.type === "ObjectExpression") {
+			objectExp = p.value.value;
+		}
+		// object -> loop through it
+		Object.keys(value).forEach(prop => {
+			addProperty(j, objectExp, prop, value[prop]);
+		});
+		valForNode = objectExp;
+	} else {
+		valForNode = createIdentifierOrLiteral(j, value);
+	}
+	let pushVal;
+	if (key) {
+		pushVal = j.property("init", j.identifier(key), valForNode);
```

Comment:
When do you have this case?

Reply:
Almost every case. What happens is that we compute the right-hand part of a particular property (for instance _development_ or _production_) and store it in `valForNode` variable. We then push this value to the key (_mode_ in this case). Similarly, `valForNode` can be a Literal (_mode_), ArrayExpression (_plugins_) or ObjectExpression(_output_).

The else part of this check happens when we recursively add something. For instance, we have an array of objects (_rules_). We will recursively add every object to an ArrayExpression. In this case, there is no `key`. We just have to keep adding objects inside the array.

Reply:
Thanks :smile: